src/test/java/data/TaskManagerExceptionTest.java

```diff
@@ -31,6 +34,89 @@ public void checkIfDateInCurrentMonth_pastDateGiven_exceptionThrown () {
                 thrown.getMessage());
     }
 
+    @Test
+    public void checkIfTimeInFormat_invalidTimeGiven_exceptionThrown () {
+        TaskManagerException thrown = Assertions.assertThrows(TaskManagerException.class, () ->
+                checkIfTimeInFormat("abcde"),
+                "TaskManagerException was expected");
+
+        Assertions.assertEquals("Invalid time format. Please use the format HHmm", thrown.getMessage());
+    }
+
+    @Test
+    void checkIfTimeInFormat_invalidFormat_throwsException() {
+        String invalidTime = "25:00";
+        assertThrows(TaskManagerException.class, () -> TaskManagerException.checkIfValidTime(invalidTime));
+    }
+
+    @Test
+    void checkIfTimeInFormat_validFormat_noExceptionThrown() {
+        String validTime = "23:59";
+        try {
+            TaskManagerException.checkIfValidTime(validTime);
+        } catch (TaskManagerException e) {
+            assert false;
+        }
+    }
+
+    @Test
+    void checkIfTimeInFormat_invalidMinute_throwsException() {
+        String invalidTime = "23:60";
+        assertThrows(TaskManagerException.class, () -> TaskManagerException.checkIfValidTime(invalidTime));
+    }
```

Comment:
Good job on adding extensive tests and also on testing both positive and negative test cases!